src/webgl/p5.RendererGL.js

```diff
@@ -830,6 +780,122 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
       this._pInst,
       !isPGraphics
     );
+
+
+    const lightingShader = renderer.getCachedShader('lightingShader',readFileSync(
+      join(__dirname, '/shaders/lighting.glsl'),
+      'utf-8'
+    ));
+    const webgl2CompatibilityShader = readFileSync(
+      join(__dirname, '/shaders/webgl2Compatibility.glsl'),
+      'utf-8'
+    );
+
+    defaultShaders = {
+      immediateVert: renderer.getCachedShader('immediateVert',readFileSync(
```

Comment:
One issue with this current structure is that we're calling `getCachedShader` in the `in `_resetContext`, so on load, it will try to make a bunch of shaders.

We maybe want to keep the shader source code in objects outside the constructor as it was before, and then we only call `getCachedShader` when we need to *use* the shader. e.g.:
```js
  _getCubemapShader() {
    return this.getCachedShader(
      'defaultCubemapShader', 
      this._webGL2CompatibilityPrefix('vert', 'mediump') +
        defaultShaders.cubemapVertexShader,
      this._webGL2CompatibilityPrefix('frag', 'mediump') +
        defaultShaders.cubemapFragmentShader
    )
  }
```